### PR TITLE
SDL module of ATF improved to provide SDL build options to test scripts.

### PR DESCRIPTION
**Pull request by aderiabin**
-  SDL module of ATF improved to provide SDL build options to test scripts.
 - Test base module was updated to handle skipping of tests

### CODE REVIEW

File: modules/SDL.lua [hunk 1]
```diff
@@ -1,15 +1,60 @@
...
+  return nil
+end
+
+local function setSdlBuildOption(optionName, sdlBuildParam, defaultValue)
+  local value = readParameterFromCMakeCacheFile(sdlBuildParam)
+  if value == nil then
+    value = defaultValue
+    local msg = "SDL build option " ..
+      sdlBuildParam .. " are unavailable.\nAssume that SDL was built with " ..
+      sdlBuildParam .. " is " .. defaultValue
```
Comment (AStasiuk) [inline modules/SDL.lua]: Think it's better to use '=' instead of 'is'.

File: modules/SDL.lua [hunk 2]
```diff
@@ -1,15 +1,60 @@
...
+  end
+  return nil
+end
+
+local function setSdlBuildOption(optionName, sdlBuildParam, defaultValue)
+  local value = readParameterFromCMakeCacheFile(sdlBuildParam)
+  if value == nil then
+    value = defaultValue
+    local msg = "SDL build option " ..
+      sdlBuildParam .. " are unavailable.\nAssume that SDL was built with " ..
```
[the remark this answers is not in the document]
Comment (aderiabin) [reply modules/SDL.lua]: @AStasiuk Fixed in 1a1ff45

File: modules/SDL.lua (hunk 1 above)
Comment (aderiabin) [reply modules/SDL.lua]: @AStasiuk Fixed in 1a1ff45

File: modules/SDL.lua [hunk 3]
```diff
@@ -81,4 +126,6 @@ function SDL:DeleteFile()
   end
 end
 
+setAllSdlBuildOptions()
```
Comment (dboltovskyi) [inline modules/SDL.lua]: @aderiabin, This function is executed as many times as module SDL "required"
For now this module is required in testbase and connecttest (dummy_connect_test, hmi_values etc.)

So need some logic to execute this function only once

File: modules/SDL.lua [hunk 4]
```diff
@@ -1,15 +1,67 @@
...
+    defaultValue = "ON"
+  },
+  extendedPolicy =  {
+    sdlBuildParameter = "EXTENDED_POLICY",
+    defaultValue = "PROPRIETARY"
+  }
+}
+
+local function readParameterFromCMakeCacheFile(paramName)
+  local pathToFile = config.pathToSDL .. "/CMakeCache.txt"
```
Comment (LuxoftAKutsan) [inline modules/SDL.lua]: @aderiabin where is our wonderful function path.join?

File: modules/SDL.lua [hunk 5]
```diff
@@ -1,15 +1,67 @@
...
+    sdlBuildParameter = "REMOTE_CONTROL",
+    defaultValue = "ON"
+  },
+  extendedPolicy =  {
+    sdlBuildParameter = "EXTENDED_POLICY",
+    defaultValue = "PROPRIETARY"
+  }
+}
+
+local function readParameterFromCMakeCacheFile(paramName)
```
Comment (LuxoftAKutsan) [inline modules/SDL.lua]: @aderiabin lack of decription

File: modules/SDL.lua [hunk 6]
```diff
@@ -1,15 +1,67 @@
...
+      paramType, paramValue = string.match(line, "^%s*" .. paramName .. ":(.+)=(%S*)")
+      if paramValue then
+        return paramValue, paramType
+      end
+    end
+  end
+  return nil
+end
+
+local function setSdlBuildOption(optionName, sdlBuildParam, defaultValue)
```
Comment (LuxoftAKutsan) [inline modules/SDL.lua]: @aderiabin lack of description

File: modules/SDL.lua [hunk 7]
```diff
@@ -1,15 +1,67 @@
...
+    print(console.setattr(msg, "cyan", 1))
+  else
+    if paramType == "UNINITIALIZED" then
+      value = nil
+      local msg = "SDL build option " ..
+        sdlBuildParam .. " is unsupported."
+      print(console.setattr(msg, "cyan", 1))
+    end
+  end
+  SDL.buildOptions[optionName] = value
```
Comment (LuxoftAKutsan) [inline modules/SDL.lua]: @aderiabin result of this method is a modification of SDL table. It is very not obvious. You could make this function as member SDL.

File: modules/SDL.lua [hunk 8]
```diff
@@ -1,15 +1,67 @@
...
+end
+
+local function setAllSdlBuildOptions()
```
Comment (LuxoftAKutsan) [inline modules/SDL.lua]: @aderiabin lack of description, and not obvious that raw function changes SDL table. Can you make it member?

File: modules/SDL.lua (hunk 5 above)
Comment (aderiabin) [reply modules/SDL.lua]: @LuxoftAKutsan  Description in LDoc format was added (see 5aec3b3)

File: modules/SDL.lua (hunk 6 above)
Comment (aderiabin) [reply modules/SDL.lua]: @LuxoftAKutsan  Description in LDoc format was added (see 5aec3b3)

File: modules/SDL.lua (hunk 7 above)
Comment (aderiabin) [reply modules/SDL.lua]: @LuxoftAKutsan  Description in LDoc format was added (see 5aec3b3)
SDL module as parameter for local function was added (see b5835e5)

File: modules/SDL.lua (hunk 8 above)
Comment (aderiabin) [reply modules/SDL.lua]: @LuxoftAKutsan  Description in LDoc format was added (see 5aec3b3)
SDL module as parameter for local function was added (see b5835e5)

**Pull request merged**